Description: Upstream changes introduced in version 5.22.0-1
 This patch has been created by dpkg-source during the package build.
 Here's the last changelog entry, hopefully it gives details on why
 those changes were made:
 .
 maxima (5.22.0-1) unstable; urgency=high
 .
   * New upstream release
   * Bug fix: "package purge (after dependencies removal) fails", thanks to
     Lucas Nussbaum (Closes: #591862).
   * Bug fix: "Ubuntu changes to maxima-emacs installation", thanks to
     Stefano Rivera (Closes: #586563).
 .
 The person named in the Author field signed this changelog entry.
Author: Camm Maguire <camm@debian.org>
Bug-Debian: http://bugs.debian.org/586563
Bug-Debian: http://bugs.debian.org/591862

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- maxima-5.22.0.orig/interfaces/emacs/misc/bookmode.el
+++ maxima-5.22.0/interfaces/emacs/misc/bookmode.el
@@ -680,7 +680,7 @@ file")
 		
 		  (and tem
 		       (list (concat "View "(nth 1 tem))
-			     'call-process  "ghostview" nil nil nil
+			     'call-process  "gv" nil nil nil
 			     (expand-file-name(nth 1 tem))
 			     ))
 		  (list "Cancel")
@@ -705,7 +705,7 @@ file")
 			 (concat
 			  (cond ((looking-at "%PS") "")
 				(t "gzip -dc | "))
-			  "ghostview -")
+			  "gv -")
 			 
 	)))
 	 
--- maxima-5.22.0.orig/interfaces/emacs/emaxima/maxima.el
+++ maxima-5.22.0/interfaces/emacs/emaxima/maxima.el
@@ -2495,7 +2495,7 @@ if completion is ambiguous."
     (define-key map "\C-c\)" 'maxima-check-parens)
     (define-key map [(control c) (control \))] 'maxima-check-form-parens)
 ;    (define-key map "\C-cC-\)" 'maxima-check-form-parens)
-    (define-key map "\177" 'backward-delete-char-untabify)
+;    (define-key map "\177" 'backward-delete-char-untabify)
     (setq maxima-mode-map map)))
 
 ;;;; Menu
@@ -3317,7 +3317,7 @@ To scroll through previous commands,
 (define-key inferior-maxima-mode-map [(meta control tab)] 
   'inferior-maxima-input-complete)
 (define-key inferior-maxima-mode-map "\e\t" 'inferior-maxima-complete)
-(define-key inferior-maxima-mode-map "\177" 'backward-delete-char-untabify)
+;(define-key inferior-maxima-mode-map "\177" 'backward-delete-char-untabify)
 (define-key inferior-maxima-mode-map "\C-c\C-k" 'maxima-stop)
 (define-key inferior-maxima-mode-map "\C-c\C-d" maxima-help-map)
 
--- maxima-5.22.0.orig/interfaces/xmaxima/Tkmaxima/Constants.tcl
+++ maxima-5.22.0/interfaces/xmaxima/Tkmaxima/Constants.tcl
@@ -148,7 +148,7 @@ set maxima_priv(urlHandlers) {
     image/gif  netmath
     image/png  netmath
     image/jpeg netmath
-    application/postscript "ghostview -safer %s"
+    application/postscript "gv -safer %s"
     application/pdf "acroread %s"
     application/x-dvi "xdvi %s"
 }
--- maxima-5.22.0.orig/src/init-cl.lisp
+++ maxima-5.22.0/src/init-cl.lisp
@@ -521,8 +521,9 @@ When one changes, the other does too."
     ;; Autoload for Maxima documantation index file
     (let ((subdir-bit (if (null *maxima-lang-subdir*) "." *maxima-lang-subdir*)))
       ;; Assign AUTOLOAD property instead of binding a function (the result of AUTOF).
-      (setf (get 'cl-info::cause-maxima-index-to-load 'autoload)
-	    (combine-path *maxima-infodir* subdir-bit "maxima-index.lisp")))))
+      (unless (get 'cl-info::cause-maxima-index-to-load 'autoload)
+	(setf (get 'cl-info::cause-maxima-index-to-load 'autoload)
+	    (combine-path *maxima-infodir* subdir-bit "maxima-index.lisp"))))))
 
 (defun get-dirs (path)
   #+(or :clisp :sbcl :ecl :openmcl)
--- maxima-5.22.0.orig/lisp-utils/defsystem.lisp
+++ maxima-5.22.0/lisp-utils/defsystem.lisp
@@ -4671,6 +4671,7 @@ the system definition, if provided."
 			  source-pname
 			  :output-file
 			  output-file
+			  #+gcl :system-p #+gcl t
 			  #+(or :cmu :scl) :error-file
 			  #+(or :cmu :scl) (and *cmu-errors-to-file*
 						(component-full-pathname component
